source/crud/tests/unified/estimatedDocumentCount.yml

```
@@ -265,3 +265,65 @@ tests:
                 count: *collection0Name
               commandName: count
               databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with comment on 4.9.0 or greater"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          comment: &objectComment { key: "value"}
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0Name
+                pipeline: *pipeline
+                comment: *objectComment
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with comment on less than 4.9.0"
+    runOnRequirements:
+      # https://jira.mongodb.org/browse/SERVER-63315
+      # Server supports count with comment of any type for comment starting from 4.4.14.
+      - minServerVersion: "4.4.14"
+        maxServerVersion: "4.8.99"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          comment: *objectComment
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                count: *collection0Name
+                comment: *objectComment
+              commandName: count
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with comment on less than 4.4.14 - string comment"
+    runOnRequirements:
+      - minServerVersion: "4.4.0"
+        maxServerVersion: "4.4.13"
```

```suggestion
  - description: "estimatedDocumentCount with string comment - pre 4.9"
    runOnRequirements:
      - minServerVersion: "4.4.0"
        maxServerVersion: "4.8.99"
```

Assuming 4.4.0 is the first server version where `count` accepts a `comment` option at all, I don't think we need such a restrictive upper bound on this test. Typically, we'd have a test for both string and document types, with document filling in for _all_ non-string BSON types.

---

Done